best_line.h

```diff
@@ -0,0 +1,52 @@
+#ifndef BEST_LINE_H
+#define BEST_LINE_H
+
+#include <float.h>
+#include <limits.h>
+
+#include <iostream>
+#include <unordered_map>
+#include <vector>
+
+using namespace std;
+
+/*Object to represent a point with x,y coordinates*/
+struct Point {
```

Comment:
This code is correct, but if you want to keep it this simple (which is fine)  I would 
* initialize data members at declaration with {}
* drop the CTOR

Then, you can initialize an object with {1.1, 2.2} instead of Point(1.1, 2.2)

This is a question of style though.